### PR TITLE
Define endpoint state

### PR DESCRIPTION
**Pull request by sawsa307**
Define SyncerEPStat to store the count of endpoint and endpoint slices in different states. Define the states for endpoint and endpoint slices.

### CODE REVIEW

File: pkg/neg/metrics/neg_metrics_collector.go [hunk 1]
```diff
@@ -66,3 +74,20 @@ func (sm *SyncerMetrics) Run(stopCh <-chan struct{}) {
 // export exports syncer metrics.
 func (sm *SyncerMetrics) export() {
```
Comment (swetharepakula) [inline pkg/neg/metrics/neg_metrics_collector.go:75]: Should there be a change her to export these metrics?

File: pkg/neg/types/endpoint_stat.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+)
+
+func StateForEP() []State {
+	return []State{EPMissingNodeName, EPMissingPod, EPMissingZone, EPMissingField, EPDuplicate, EPTotal}
+}
+
+func StateForEPS() []State {
+	return []State{EPSWithMissingNodeName, EPSWithMissingPod, EPSWithMissingZone, EPSWithMissingField, EPSWithDuplicate, EPSTotal}
+}
+
```
Comment (swetharepakula) [inline pkg/neg/types/endpoint_stat.go:48]: What are these functions used for?

File: pkg/neg/types/endpoint_stat.go [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package types
+
+type State string
+
+func (s State) String() string {
```
Comment (swetharepakula) [inline pkg/neg/types/endpoint_stat.go:21]: Why is this necessary?

File: pkg/neg/metrics/neg_metrics_collector.go (hunk 1 above)
Comment (sawsa307) [reply pkg/neg/metrics/neg_metrics_collector.go:75]: This PR is for definition of variables. Metrics and export will come in a separate PR. Thanks!

File: pkg/neg/types/endpoint_stat.go (hunk 3 above)
Comment (sawsa307) [reply pkg/neg/types/endpoint_stat.go:21]: Removed. Thanks!

File: pkg/neg/types/endpoint_stat.go (hunk 2 above)
Comment (sawsa307) [reply pkg/neg/types/endpoint_stat.go:48]: They are used to make sure we go over each state when we export. I'll remove them for now and address this in the metrics PR. Thanks!

File: pkg/neg/types/endpoint_stat.go [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package types
+
+type State string
+
+const (
```
Comment (swetharepakula) [inline pkg/neg/types/endpoint_stat.go:21]: nit: can you capitalize all of the states?

Comment (sawsa307) [reply pkg/neg/types/endpoint_stat.go:21]: Updated. Thank you!

**Review by swetharepakula — COMMENTED** [review]
/lgtm
/approve
